lib/utils.js

```diff
@@ -53,3 +53,31 @@ export async function loadGetInitialProps (Component, ctx) {
   }
   return props
 }
+
+export class LockManager {
```

Comment:
@nkzawa But then we've to wait all the jobs getting done no? Usually we are getting requests by one by one. 
Then we may need to do a setTimeout and check all the prefetches happening in the same event loop and so on.

But I thought this was nice than that.